pkg/asset/installconfig/azure/client.go

```diff
@@ -152,3 +154,29 @@ func (c *Client) getProvidersClient(ctx context.Context) (azres.ProvidersClient,
 	client.Authorizer = c.ssn.Authorizer
 	return client, nil
 }
+
+// GetResourceSku checks if the diskType exists for the given region.
+func (c *Client) GetResourceSku(ctx context.Context, region string, diskType string) (*azsku.ResourceSku, error) {
+	client := azsku.NewResourceSkusClient(c.ssn.Credentials.SubscriptionID)
+	client.Authorizer = c.ssn.Authorizer
+	ctx, cancel := context.WithTimeout(ctx, 30*time.Second)
+	defer cancel()
+
+	resourceSkus, err := client.List(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, page := range resourceSkus.Values() {
+		if *page.Name == diskType {
+			for _, diskRegion := range *page.Locations {
+				if diskRegion == region {
+					return &page, nil
+				}
+			}
+			return nil, errors.New("ultraSSD_LRS is not available in specified region")
+		}
+	}
```

Comment:
1. this is only listing one page. there might be a lot more.

Reply:
1. use autorest/to for ptr to string like changes.

Reply:
> azsku "github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2018-10-01/compute"

are you sure the compute list skus returns disk SKUs ?? my `az vm list-skus -otable` is returning only virtual machine types skus..

Reply:
>are you sure the compute list skus returns disk SKUs ?? my az vm list-skus -otable is returning only virtual machine types skus..

I think that the ResourceSkuClient calls [this endpoint](https://docs.microsoft.com/en-us/rest/api/compute/resourceskus/list) to get the list of Resource SKUs available for the subscription. Not sure if they are only virtual machine types. While testing, I did get a list of disks but did not verify if they belong to only Virtual Machine types skus.

Reply:
It looks like you can use https://godoc.org/github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2018-10-01/compute#ResourceSkusResultPage.NextWithContext to advance the pages

Reply:
ah just got the values in page and multiple pages reference. Making the changes.